Use forwardable logging in WebRTC WebProcess code

youennf opened the pull request:
#### bd1e24d04518411e0543fc9b37239eb944b2030e
<pre>
Use forwardable logging in WebRTC WebProcess code
<a href="https://rdar.apple.com/157117303">rdar://157117303</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=296701">https://bugs.webkit.org/show_bug.cgi?id=296701</a>

Reviewed by Eric Carlson and Per Arne Vollan.

We make sure that logs that can be very long are forwarded properly so that they are not shortened by the default log forwarding mechanism.
We apply this to WebRTC stats, libwebrtc messages and SDP logging.
We introduce a new Logger method so that the above logging can keep being seen in web inspector.

As a side fix, we no longer convert to hexa log identifiers given we sometimes log them as pointers, which end up being printed as integers.
We are simply printing them as integers to make it easier to relate the logs.

* Source/WTF/wtf/Logger.cpp:
(WTF::Logger::LogSiteIdentifier::toString const):
(WTF::&gt;::toString):
* Source/WTF/wtf/Logger.h:
(WTF::Logger::toObservers const):
(WTF::Logger::Logger):
(WTF::Logger::log):
(WTF::Logger::sendMessageToObservers):
(WTF::Logger::logVerbose):
* Source/WebCore/Modules/mediastream/PeerConnectionBackend.cpp:
(WebCore::PeerConnectionBackend::createOfferSucceeded):
(WebCore::PeerConnectionBackend::createOfferFailed):
(WebCore::PeerConnectionBackend::createAnswerSucceeded):
(WebCore::PeerConnectionBackend::createAnswerFailed):
* Source/WebCore/Modules/mediastream/RTCPeerConnection.cpp:
(WebCore::RTCPeerConnection::setLocalDescription):
(WebCore::RTCPeerConnection::setRemoteDescription):
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCMediaEndpoint.cpp:
(WebCore::LibWebRTCMediaEndpoint::OnStatsDelivered):
* Source/WebCore/platform/LogMessages.in:
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProvider.cpp:
(WebCore::doReleaseLogging):

Originally-landed-as: 06d8dd827285. <a href="https://rdar.apple.com/157117303">rdar://157117303</a>
Canonical link: <a href="https://commits.webkit.org/298889@main">https://commits.webkit.org/298889@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7b1641a6d48b57faba7ee29caafdb67f26f64b98

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/116979 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/36643 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/27225 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/123063 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/68996 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/0c7ac1db-12da-4a59-b562-b1ab395a736a) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/118867 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/37345 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/45244 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/88847 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/43561 "") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/40edd5f1-a5c3-4c04-a288-cdcbe1544009) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/119927 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/29784 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/104947 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/69308 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/28852 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/23054 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/66719 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/109095 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/99174 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/23207 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/126194 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/115506 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/43881 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/32984 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/97517 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/44243 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/101148 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/97317 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24774 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/42643 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/20584 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/40273 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/43763 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/49363 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/144205 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/43230 "Built successfully") | | [💥 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/37129 "An unexpected error occured. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/46572 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/44937 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3c7c70170953c9555b5f6c42af769b8a40297a6d. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48725

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/06d8dd8272853656a994c4fbc15a6f03ebe898e2. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48725

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/7b1641a6d48b57faba7ee29caafdb67f26f64b98. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48725

youennf committed:
Use forwardable logging in WebRTC WebProcess code
rdar://157117303
https://bugs.webkit.org/show_bug.cgi?id=296701

Reviewed by Eric Carlson and Per Arne Vollan.

We make sure that logs that can be very long are forwarded properly so that they are not shortened by the default log forwarding mechanism.
We apply this to WebRTC stats, libwebrtc messages and SDP logging.
We introduce a new Logger method so that the above logging can keep being seen in web inspector.

As a side fix, we no longer convert to hexa log identifiers given we sometimes log them as pointers, which end up being printed as integers.
We are simply printing them as integers to make it easier to relate the logs.

* Source/WTF/wtf/Logger.cpp:
(WTF::Logger::LogSiteIdentifier::toString const):
(WTF::>::toString):
* Source/WTF/wtf/Logger.h:
(WTF::Logger::toObservers const):
(WTF::Logger::Logger):
(WTF::Logger::log):
(WTF::Logger::sendMessageToObservers):
(WTF::Logger::logVerbose):
* Source/WebCore/Modules/mediastream/PeerConnectionBackend.cpp:
(WebCore::PeerConnectionBackend::createOfferSucceeded):
(WebCore::PeerConnectionBackend::createOfferFailed):
(WebCore::PeerConnectionBackend::createAnswerSucceeded):
(WebCore::PeerConnectionBackend::createAnswerFailed):
* Source/WebCore/Modules/mediastream/RTCPeerConnection.cpp:
(WebCore::RTCPeerConnection::setLocalDescription):
(WebCore::RTCPeerConnection::setRemoteDescription):
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCMediaEndpoint.cpp:
(WebCore::LibWebRTCMediaEndpoint::OnStatsDelivered):
* Source/WebCore/platform/LogMessages.in:
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProvider.cpp:
(WebCore::doReleaseLogging):

Originally-landed-as: 06d8dd827285. rdar://157117303
Canonical link: https://commits.webkit.org/298889@main
